### DISCUSSION TITLE
Sketch uploading under Linux

### DISCUSSION THREAD

**Discussion by tbillion** [earlephilhower/arduino-pico#913] [General]
im on linux, 1.8 ide still, recently i upgraded from 2.3.2 library to the 2.5.x library and then to the 2.6.x library i up graded to 2.6 because 2.5.x started casuing this issue, however the isue persists after the down grade back to 2.3.2 . 

my issue is that after programming if programming is even successful, the serial port never shows back up on the laptop..i have to hold down the white button to program it and then sometimes it doesnt work. however if i switch to the mbed core everything goes back to functioning as it should. However the program i have wrote that is several thousand lines needs the Philhower core to access features that are locked out by mbed. im not sure how to diagnose this help would be appreciated. 

it does it with simple and complicated sketches alike this is the diag from blink. 

Sketch uses 49252 bytes (2%) of program storage space. Maximum is 2093056 bytes.
Global variables use 7244 bytes (2%) of dynamic memory, leaving 254900 bytes for local variables. Maximum is 262144 bytes.
Resetting /dev/ttyACM1
Converting to uf2, output size: 104960, start address: 0x2000
No drive to deploy.
An error occurred while uploading the sketch

**Edit, it only affects when the clock speed is on the default 133MHZ, if imove it to 100, i get one good upload that will sort of function. if i change it back again then i get another half way functioning pico... its weird.

also if by some means i get a philhower core program to load you cant go back and upload a mbed core sketch either it always fails to upload.

**Comment by earlephilhower** [comment]
Moving to a discussion because this is not really core code related and there are some general things anyone having this problem shoulw look at:

* What distro/version are you using?
* What RP2040 board, exactly?
* Have you selected the proper serial port?  Does it work in Serial Monitor before upload?
* Is the Pico crashed?  You need to use BOOTSEL if your app crashes since the USB port is controlled by the RP2040
* Are you using a `snap`?  If so, delete it and use the real app because of permissions problems.
* Does the board reset and show up in a list of filesystems?  Where?
* When you hold the BOOTSEL and plug in, does it appear in the list of FS?  Where?
* Have you tried other boards and/or other cables?
